packages/model-viewer/src/test/model-viewer-spec.ts

```diff
@@ -0,0 +1,136 @@
+import {ModelViewerElement} from '../model-viewer.js';
+import {Constructor} from '../utilities.js';
+
+import {assetPath, FidelityResult, SCENARIOS, waitForEvent} from './helpers.js';
+import {BasicSpecTemplate} from './templates.js';
+
+const expect = chai.expect;
+
+const setupModelViewer =
+    async (modelViewer: ModelViewerElement, config: any) => {
+  modelViewer.style.width = `${config.dimensions.width}px`;
+  modelViewer.style.height = `${config.dimensions.height}px`;
+
+  modelViewer.style.backgroundColor = 'rgba(255,255,255,0)';
+
+  const {model, lighting, orbit, renderSkybox} = config;
+  modelViewer.src = assetPath(model);
+
+  const lightingPath = assetPath(lighting);
+  modelViewer.environmentImage = lightingPath;
+  modelViewer.skyboxImage = renderSkybox ? lightingPath : null;
+
+  const {theta, phi, radius} = orbit;
+  modelViewer.minCameraOrbit = `auto auto ${radius}m`;
+  modelViewer.maxCameraOrbit = `auto auto ${radius}m`;
+  modelViewer.cameraOrbit = `${theta}deg ${phi}deg ${radius}m`;
+  const {x, y, z} = config.target;
+  modelViewer.cameraTarget = `${x}m ${y}m ${z}m`;
+  modelViewer.fieldOfView = `${config.verticalFoV}deg`;
+
+  await waitForEvent(modelViewer, 'poster-dismissed');
+};
+
+async function captureScreenshot(blob: Blob): Promise<HTMLCanvasElement> {
+  const image = await new Promise<HTMLImageElement>((resolve) => {
+    const image = new Image();
+    const url = URL.createObjectURL(blob);
+    image.src = url;
+    image.onload = () => {
+      resolve(image);
+    }
+  });
+
+  const fiddleCanvas = document.createElement('canvas');
+  fiddleCanvas.width = image.width;
+  fiddleCanvas.height = image.height;
+  fiddleCanvas.getContext('2d')?.drawImage(
+      image, 0, 0, image.width, image.height);
+
+  return Promise.resolve(fiddleCanvas);
+};
+
+// TODO(sun765): this only test whether the screenshot
+// is colorless or not. Replace this with more robust
+// test in later pr.
+function testFidelity(screenshotCanvas: HTMLCanvasElement): FidelityResult {
+  let colorlessPixelCount = 0;
+  let transparentPixelCount = 0;
```

Comment:
Since i have used a different way to capture screenshot, the poster bug probably wont exist, so do i still need to check whether the background is semi-transparent?

Reply:
Yeah, that makes sense; let's skip the transparent count.